hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java

```diff
@@ -263,15 +264,19 @@ public void test_whenEntryIsUpdatedBackupIsNotCleaned() {
     @Test
     public void test_whenEntryIsRemovedBackupIsCleaned() {
         SimpleExpiryListener listener = new SimpleExpiryListener();
-        CacheConfig<Integer, Integer> cacheConfig = createCacheConfig(new HazelcastExpiryPolicy(FIVE_SECONDS, FIVE_SECONDS, FIVE_SECONDS), listener);
+        int ttlSeconds = 10;
+        Duration duration = new Duration(TimeUnit.SECONDS, ttlSeconds);
+        HazelcastExpiryPolicy expiryPolicy = new HazelcastExpiryPolicy(duration, duration, duration);
+        CacheConfig<Integer, Integer> cacheConfig = createCacheConfig(expiryPolicy, listener);
         Cache<Integer, Integer> cache = createCache(cacheConfig);
 
         for (int i = 0; i < KEY_RANGE; i++) {
             cache.put(i, i);
-            cache.remove(i);
+            assertTrue("Expected to remove entry " + i + " but entry was not present. Expired entry count: "
```

Comment:
Just an idea - if we will count how many entries cannot be removed here (instead of assertion) then we can check that number of `listener.getExpirationCount().get()` on line https://github.com/hazelcast/hazelcast/pull/13834/files#diff-f0263d0ec216029eff73800983006069R280 equals to number of entries which cannot be removed from cache (instead of check `0`) - i.e. how many of them was removed due to expiration. WDYT?

Reply:
I'm not sure if expirations will remove entries from the backup so we might need to adapt even the assertion for backup entries. If the test persists to fail, I'd redo the whole test. For now I've taken the easy approach.